src/overlays/actors/ovl_En_Zot/z_en_zot.c

```diff
@@ -22,123 +57,1366 @@ const ActorInit En_Zot_InitVars = {
+        if (gSaveContext.weekEventReg[29] & 8) {
+            textId = 0x1264;
+        } else {
+            gSaveContext.weekEventReg[29] |= 8;
+        }
+    }
+    func_801518B0(globalCtx, textId, &this->actor);
+}
+
+void func_80B973BC(EnZot* this, GlobalContext* globalCtx) {
+    func_80B96D4C(this);
+    Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 2, 0x800, 0x100);
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    if ((func_80152498(&globalCtx->msgCtx) == 5) && func_80147624(globalCtx)) {
+        switch (globalCtx->msgCtx.unk11F04) {
+            case 0x126E:
+            case 0x1270:
+            case 0x1273:
+            case 0x1274:
+                func_80151938(globalCtx, globalCtx->msgCtx.unk11F04 + 1);
+                break;
+
+            case 0x1272:
+                func_80151938(globalCtx, 0x126F);
+                break;
+
+            case 0x126F:
+                func_801159EC(90);
+                func_80151938(globalCtx, 0x1270);
+                break;
+
+            case 0x1275:
+                if (gSaveContext.rupees < 10) {
+                    func_80151938(globalCtx, 0x1277);
+                } else {
+                    func_80151938(globalCtx, 0x1278);
+                    func_801159EC(-10);
+                }
+                break;
+
+            case 0x1276:
+                func_80151938(globalCtx, 0x1275);
+                break;
+
+            case 0x1271:
+            case 0x1277:
+            case 0x1278:
+            case 0x1279:
+                func_801477B4(globalCtx);
+                func_80B965D0(this, globalCtx);
+                this->actor.flags &= ~0x10000;
+                this->actor.textId = 0;
+                this->actionFunc = func_80B97708;
+                if ((this->actor.cutscene != -1) && !(this->unk_2F2 & 1)) {
+                    ActorCutscene_Stop(this->actor.cutscene);
+                }
+                this->unk_2F2 &= ~1;
+                break;
+        }
+    }
+
+    if (this->unk_2F2 & 1) {
+        if (this->actor.cutscene == -1) {
+            this->unk_2F2 &= ~1;
+        } else if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+            ActorCutscene_Stop(0x7C);
+            ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        } else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+            ActorCutscene_StartAndSetUnkLinkFields(this->actor.cutscene, &this->actor);
+            this->unk_2F2 &= ~1;
+        } else {
+            ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        }
+    }
+}
+
+void func_80B975F8(EnZot* this, GlobalContext* globalCtx) {
+    func_80B96D4C(this);
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        this->actionFunc = func_80B973BC;
+    } else {
+        func_800B8500(&this->actor, globalCtx, 10000.0f, 1000.0f, 0);
+    }
+}
+
+void func_80B9765C(EnZot* this, GlobalContext* globalCtx) {
+    if (1) {
+        do { } while (0); }
```

Comment:
Did you forgot to run the formatter? Or the formatter thought this was a good idea?

Reply:
This is the formatter's work.